### PR TITLE
feat: implement solution to 345. Reverse vowels of a string

### PR DESCRIPTION
**Pull request by codyconway77**

### CODE REVIEW

**Review by jmlawrence — APPROVED** [review]
Awesome solution!!!

File: JS/345. Reverse Vowels of a String.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * @param {string} s
+ * @return {string}
+ */
+ var reverseVowels = function(s) {
+    const dict = {};
+    const vowels = new Set(['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']);
+    // Loop through the string and store all of the vowels along with their index
+    for (let i = 0; i < s.length; i++) {
```
Comment (jmlawrence) [inline JS/345. Reverse Vowels of a String.js]:
```suggestion
    for (let i in s) {
```

File: JS/345. Reverse Vowels of a String.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    let vowelStack = [];
+    let indexStack = [];
+    // Create inversed arrays of indecies and the vowels that will go there
+    for (index in dict) {
+        vowelStack.push(dict[index]);
+        indexStack.unshift(index);
+    }
+    let strArr = s.split('');
+    // Place all vowels in their reversed order
+    for (let i = 0; i < vowelStack.length; i++) {
```
Comment (jmlawrence) [inline JS/345. Reverse Vowels of a String.js]:
```suggestion
for (let i in vowelStack) {
```

File: JS/345. Reverse Vowels of a String.js (hunk 1 above)
Comment (codyconway77) [reply JS/345. Reverse Vowels of a String.js]: Great suggestion. Commited this to be more concise

File: JS/345. Reverse Vowels of a String.js (hunk 2 above)
Comment (codyconway77) [reply JS/345. Reverse Vowels of a String.js]: Ditto above comment

**Pull request merged**